### PR TITLE
Valkyrie walking in Drake

### PR DESCRIPTION
**Pull request by rdeits**
This is updated from #1516, mostly to remove duplication of the controllers and other functions. It also has some bug fixes and cleanups on top of that pull request.

### CODE REVIEW

**Comment by rdeits** [comment]
Okay, I've cleaned this up a bunch, and I think it's ready to go once the tests come back.

File: drake/examples/Pendulum/CMakeLists.txt [hunk 1]
```diff
@@ -11,7 +11,7 @@ include_directories(${PROJECT_SOURCE_DIR}/util)
 if (LCM_FOUND)
   include_directories(${PROJECT_SOURCE_DIR}/pod-build/lcmgen)
   add_executable(runPendulumDynamics runDynamics.cpp)
-  target_link_libraries(runPendulumDynamics drakeRBM drakeLCMSystem)
+  target_link_libraries(runPendulumDynamics drakeRBM drakeLCMSystem pthread)
```
Comment (RussTedrake) [inline drake/examples/Pendulum/CMakeLists.txt]: bad.  don't do that.

**Comment by RussTedrake** [comment]
otherwise, looks good.

File: drake/examples/Pendulum/CMakeLists.txt (hunk 1 above)
Comment (rdeits) [inline drake/examples/Pendulum/CMakeLists.txt]: fixed

File: drake/examples/Valkyrie/+valkyrieParams/Walking.m [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+      obj.whole_body.w_qdd(findPositionIndices(r, 'Forearm')) = .0001;
+      obj.whole_body.w_qdd(findPositionIndices(r, 'Wrist')) = .0001;
+
+      obj.whole_body.w_qdd(findPositionIndices(r, 'torso')) = .0001;
+
+      obj.whole_body.w_qdd(r.findPositionIndices('torsoRoll')) = 0.1;
+
+      %This gain was added specifically for Valkyrie.
+      % There must be something else going on as it shouldn't be needed.
+      obj.whole_body.w_qdd(findPositionIndices(r, 'lowerNeckPitch')) = .0001;
```
Comment (mauricefallon) [inline drake/examples/Valkyrie/+valkyrieParams/Walking.m]: I needed to add this control to the neck to be stable, but I don't see why it was necessary as it wasnt needed for Atlas.

**Comment by mauricefallon** [comment]
one other thing to remember is that I needed to extend the joint limits of the ankles in the current val urdf in drake for this to work. I think that the IHMC walking motion is also very close to these limits.

**Pull request merged**